waspls/src/Wasp/LSP/Handlers.hs

```diff
@@ -0,0 +1,123 @@
+module Wasp.LSP.Handlers
+  ( initializedHandler,
+    didOpenHandler,
+    didChangeHandler,
+    didSaveHandler,
+  )
+where
+
+import Control.Lens ((+~), (^.))
+import Control.Monad.Trans (lift)
+import Control.Monad.Trans.Except (throwE)
+import Data.Function ((&))
+import Data.Text (Text)
+import qualified Data.Text as T
+import Language.LSP.Server (Handlers, LspT)
+import qualified Language.LSP.Server as LSP
+import Language.LSP.Types
+import Language.LSP.Types.Lens
+import Language.LSP.VFS (virtualFileText)
+import qualified Wasp.Analyzer
+import qualified Wasp.Analyzer.AnalyzeError as WE
+import Wasp.Analyzer.Parser (Ctx (Ctx))
+import Wasp.Analyzer.Parser.SourceRegion (getRgnEnd, getRgnStart)
+import Wasp.LSP.State (HandlerM, ServerConfig, Severity (..))
+
+-- LSP notification and request handlers
+
+-- | "Initialized" notification is sent when the client is started. We don't
```

Comment:
Just as extra info - what are some common situations in which client is started? That is when we open a new project in an editor? Or when we open a new file? I guess those depend on the editor, but there are certainly some common situations, just to help paint the picture?